frontend/src/models/ui.ts

```diff
@@ -1,29 +1,57 @@
 import { createModel } from '@rematch/core'
 
-interface UIState {
+export const DEFAULT_INTERFACE = 'searching'
+
+type UIParams = { [key: string]: any }
+type UIState = UIParams & {
   connected: boolean
   uninstalling: boolean
+  scanLoading: { [interfaceName: string]: boolean }
+  scanTimestamp: { [interfaceName: string]: number }
+  scanInterface: string
+  setupBusy: boolean
+  setupAdded?: ITarget
+  setupDeletingDevice: boolean
+  setupAddingService: boolean
+  setupDeletingService?: number
+  setupServicesCount: number
+  setupServicesNew: boolean
 }
 
-const state: UIState = { connected: false, uninstalling: false }
+const state: UIState = {
+  connected: false,
+  uninstalling: false,
+  scanLoading: {},
+  scanTimestamp: {},
+  scanInterface: DEFAULT_INTERFACE,
+  setupBusy: false,
+  setupAdded: undefined,
+  setupDeletingDevice: false,
+  setupDeletingService: undefined,
+  setupAddingService: false,
+  setupServicesCount: 0,
+  setupServicesNew: true,
```

Comment:
Why not a nested object for this instead? Feels a bit clunky

Reply:
TBH I wasn't sure that redux would trigger a change event on just the one item that was changed and not unnecessarily trigger a change event on the other items if they were nested.